Improved saved HTML files

FreddieN opened the pull request:
I have updated the styling to reflect what @pikurasa describes in #1515.
See below:
![demo](https://i.imgur.com/zNJQ0ac.gif)
Before:
![before](https://i.imgur.com/17w7P7P.png)

The guide graphic is from a previous PR made for a similar issue (#1053). The artwork is from the _printBlockSVG component. There is a significant file increase from these changes. A 40kb file has gone up to 240kb. This is mostly taken up by the block artwork SVG, perhaps if the user is in advanced mode the script skips this stage so more proficient users can store more projects without using as much space.

Also, I had a go at designing the drag graphic myself before finding the other one. I think the drawn one suits the style of music blocks but here it is in case you prefer it: ![transfergraphic](https://i.imgur.com/ShpNa21.png)

Let me know what you think...

pikurasa commented:
The project code (JSON) is very important. It is what MB reads as its main file type.

The block code is more meaningful to most users.

At any rate, excited to test this out!!!

aust-n commented:
I should clarify that what I mean by 'removing' the Project Code is simply removing the visual aspect of the code. Specifying `display: none` should keep the project code in the file (and maintain the readability of the code into MB without it taking up the precious screen real estate). As well as this, it could also be logged to console (for debugging). 

It just seems very off putting for newer users and the ambiguity of 'code', whether that be JSON or the actual block code, could also be misinterpreted (my music teacher was confused at this). 

I think we should hide the 'visibility' of the JSON data to the user and populate that screen area with the drag and drop SVG. There is always the option to make this visible for advanced users but I don't think the JSON serves that much use to them. 

I would prefer to keep the save page minimal and simple but I am open for others' thoughts :)

@pikurasa What do you think?

walterbender commented:
@FreddieN  I like your new artwork better than the old on... more in keeping with the style. Two thoughts: (1) maybe label the pointers 1 and 2 to indicate the direction of the motion? and (2) maybe as a gif it will compress pretty well if you eliminate the dropshadows?

As it is, I am wary of increasing the file size so much.

FreddieN commented:
Updated guide image:
![guideimage](https://i.imgur.com/JkWyZgi.png)

101kb now going to 170kb... which is a bit better.
The guide image file is 30kb the rest is the raw SVG file of the block artwork. 

@aust-n I agree that it is mostly irrelevant for the user and it should be hidden and replaced with the guide artwork... @walterbender what do you think?

walterbender commented:
I am still on the fence about these changes, if only because of the file size.

pikurasa commented:
Ok, I just tested it. I find that a file that was originally 10kb goes up to 167kb, which is about a 17 fold increase.

These are my thoughts:
* According to Freddie, much of the increase is due to the SVG data. However, only the very top of MB code is visible. I think that either we a) cut the bottom part of SVG out somehow or b) make it worth it by adding a way to "show the rest of the MB code".

* I like that we have the guide, but I think it might be improved even more. In particular, I think the left side could be less busy with some indication that the user would drag the file (that they presumably just opened in their browser) from their filesystems.

* I think that it may be helpful to have this feature: https://github.com/sugarlabs/musicblocks/issues/1077 if for no other reason than it would be more straightforward to have the user copy code to their clipboard and paste into MB to be opened or drag-and-drop from the open window. Does this make sense to others? (happy to elaborate)

FreddieN commented:
@pikurasa 
I will look into cropping the SVG or some way of compressing it. Adding a "Show more" button would be fairly simple but I think file size is probably more important at the moment. I will also look into different file formats to see if there is any other way of getting the artwork file size smaller.

For the guide image, I will have a go at what you have suggested. This file is 30kb but perhaps it could be linked externally (so it isn't included with the file but stored on the music blocks server) this would mean the image would only display if the user was online though...
If we did take this approach we could perhaps add a GIF animation (which I was going to use originally but it came out too large):
![image](https://i.imgur.com/5UINH4D.gif)

The code paste feature seems fairly simple to implement, perhaps using a dialogue box similar to what we use for the filename of a saved file would work:
![image](https://i.imgur.com/RrZEDRs.png)
